plugin/src/main/java/org/opensearch/ml/utils/RestActionUtils.java

```diff
@@ -63,4 +66,20 @@ public static SearchQueryInputDataset buildSearchQueryInput(RestRequest request,
             );
         return new SearchQueryInputDataset(Arrays.asList(searchRequest.indices()), searchRequest.source());
     }
+
+    /**
+     * Checks to see if the request came from Kibana, if so we want to return the UI Metadata from the document.
+     * If the request came from the client then we exclude the UI Metadata from the search result.
+     *
+     * @param request rest request
+     * @return instance of {@link org.opensearch.search.fetch.subphase.FetchSourceContext}
+     */
+    public static FetchSourceContext getSourceContext(RestRequest request) {
+        String userAgent = Strings.coalesceToEmpty(request.header("User-Agent"));
+        if (!userAgent.contains(OPENSEARCH_DASHBOARDS_USER_AGENT)) {
+            return new FetchSourceContext(true, Strings.EMPTY_ARRAY, UI_METADATA_EXCLUDE);
```

Comment:
ML plugin doesn't have UI, so no need to return `ui_metadata`.

Reply:
sure will do in the next PR.